### PR TITLE
added ability to change the redstone lamp turning off delay(default 0)

### LINKED ISSUE #50: Remove the 2-tick delay of Redstone Lamps turning off
Makes lamp displays difficult, to the point some redstoners retexture Iron Trapdoors to look like Redstone Lamps, solely due to them having no end delay

### PR DESCRIPTION
**Pull request by RandmlyCoding**

### CODE REVIEW

**Comment by Flatkat** [comment]
LGTM! I would just add a tooltip saying what the default vanilla behaviour is, with the code not running if you set it to that number (to avoid issues).
Or well, even better, just make this a bool, i dont think people will be interested in tweaking this behaviour with smth more complex than a true or false, and avoids confusion, since a 0 wouldnt disable the feature as it would normally be expected in these kinds of situations.

**Comment by Flatkat** [comment]
Please, properly capitalize the text and use a tooltip instead of a comment.
For that, use the following on the lang file: `"modid.midnightconfig.name.tooltip":"I am a tooltip uwu \nI am a new line",`
For more information, visit MidnightLib's wiki: https://www.midnightdust.eu/wiki/midnightlib/

**Comment by RandmlyCoding** [comment]
oh, i missed that part of the documentation. Its fixed now

**Comment by Flatkat** [comment]
Will test the changes i made when im free to do so

**Comment by Flatkat** [comment]
Still the code may get re-reviewed soon prior to merge, or after merging. we will see

File: src/main/java/dev/symphony/harmony/mixin/redstone/ChangeRedstoneLampDelay.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+@Mixin(RedstoneLampBlock.class)
+public abstract class ChangeRedstoneLampDelay extends Block {
+
+    @Final @Shadow
+    public static BooleanProperty LIT;
+
+    public ChangeRedstoneLampDelay(Settings settings) {super(settings);}
+
+    //Set the delay to redstoneBulbDelay(default 0) in scheduleBlockTick
+    @Inject(method = "neighborUpdate",at = @At("HEAD"), cancellable = true)
```
Comment (axialeaa) [inline src/main/java/dev/symphony/harmony/mixin/redstone/ChangeRedstoneLampDelay.java:28]: Injecting at head and using `ci.cancel()` is generally unfavorable as it causes other modder's changes to completely fail to apply without any warning of a conflict.

There are several things you can do here, but my recommendation would be to use `@WrapOperation` at `"INVOKE"` targeting the call to `scheduleBlockTick()`. That call will then be passed into the arg `original`, which you can call by default if the config option is disabled.

If the config option is *enabled*, you can just call `scheduledTick()` directly. No need to copy and paste any code from inside that method, as `scheduledTick()` gets called as part of the schedule block tick process anyway.

File: src/main/java/dev/symphony/harmony/mixin/redstone/ChangeRedstoneLampDelay.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    //if the delay is 0, instead of sending it through scheduleBlockTick, which takes at least 1 tick, directly do it so its faster    
+        if(HarmonyConfig.removeRedstoneLampDelay){
+            if (!world.isClient) {
+                boolean bl = state.get(LIT);
+                if (bl != world.isReceivingRedstonePower(pos)) {
+                    world.setBlockState(pos, state.cycle(LIT), 2);
+                }
+            }
+        }
+        ci.cancel();
```
Comment (axialeaa) [inline src/main/java/dev/symphony/harmony/mixin/redstone/ChangeRedstoneLampDelay.java:40]: At any rate, this `ci.cancel()` should be moved inside the bracket. Currently it's preventing the lamp from ever changing state regardless of the option setting.

File: src/main/java/dev/symphony/harmony/mixin/redstone/ChangeRedstoneLampDelay.java (hunk 1 above)
Comment (axialeaa) [inline src/main/java/dev/symphony/harmony/mixin/redstone/ChangeRedstoneLampDelay.java]: Injecting at head and using `ci.cancel()` is generally unfavorable as it causes other modder's changes to completely fail to apply without any warning of a conflict.

There are several things you can do here, but my recommendation would be to use `@WrapOperation` at `"INVOKE"` targeting the call to `scheduleBlockTick()`. That call will then be passed into the arg `original`, which you can call by default if the config option is disabled.

If the config option is *enabled*, you can just call `scheduledTick()` directly. No need to copy and paste any code from inside that method, as `scheduledTick()` gets called as part of the schedule block tick process anyway.

File: src/main/java/dev/symphony/harmony/mixin/redstone/ChangeRedstoneLampDelay.java (hunk 2 above)
Comment (axialeaa) [inline src/main/java/dev/symphony/harmony/mixin/redstone/ChangeRedstoneLampDelay.java]: At any rate, this `ci.cancel()` should be moved inside the bracket. Currently it's preventing any of the neighbor update code from running regardless of the option setting.

**Review by axialeaa — CHANGES_REQUESTED** [review]

**Comment by axialeaa** [comment]
Sorry for the mess just then; I submitted them as single comments instead of as a review!

**Review by axialeaa — APPROVED** [review]
Excellent, looks good to me

**Comment by Flatkat** [comment]
Just tested and works on 1.21.3, no port required. Merging

**Comment by RandmlyCoding** [comment]
#20

**Pull request merged**